docs/src/main/asciidoc/cdi-integration.adoc

```diff
@@ -0,0 +1,597 @@
+            if (context.getQualifiers().stream()
+                    .anyMatch(a -> a.name().equals(DotName.createSimple(MyQualifier.class.getName())))) {
+                context.transform()
+                        .removeAll()
+                        .add(DotName.createSimple(MyOtherQualifier.class.getName()))
+                        .done();
+            }
+        }
+    });
+}
+----
+
+NOTE: In theory, you can use <<annotations_transformer_build_item,an `AnnotationsTransformer`>> to achieve the same goal. However, there are few differences that make `InjectionPointsTransformer` more suitable for this particular task: (1) Annotation transformers are applied before to all classes during bean discovery whereas `InjectionPointsTransformer` is only applied to discovered injection point after bean discovery and (2) with `InjectionPointsTransformer` you don't need to handle various types of injection points (field, parameters of initializer methods, etc.).
+
+== Use Case - Resource Annotations and Injection
+
+The `ResourceAnnotationBuildItem` can be used to specify resource annotations that make it possible to resolve non-CDI injection points, such as Java EE resources.
+An integrator must also provide a corresponding `io.quarkus.arc.ResourceReferenceProvider` service provider implementation. 
+
+.`ResourceAnnotationBuildItem` Example
+[source,java]
+----
+@BuildStep
+void setupResourceInjection(BuildProducer<ResourceAnnotationBuildItem> resourceAnnotations, BuildProducer<GeneratedResourceBuildItem> resources) {
+    resources.produce(new GeneratedResourceBuildItem("META-INF/services/io.quarkus.arc.ResourceReferenceProvider",
+        MyResourceReferenceProvider.class.getName().getBytes()));
+    resourceAnnotations.produce(new ResourceAnnotationBuildItem(DotName.createSimple(MyAnnotation.class.getName())));
+}
+----
+
+[[build_metadata]]
+== Available Build Time Metadata
+
+Any of the above extensions that operates with `BuildExtension.BuildContext` can leverage certain build time metadata that are generated during build.
+The built-in keys located in `io.quarkus.arc.processor.BuildExtension.Key` are:
+
+* `ANNOTATION_STORE`
+** Contains an `AnnotationStore` that keeps information about all `AnnotationTarget` annotations after application of annotation transformers
+* `INJECTION_POINTS`
+** `Collection<InjectionPointInfo>` containing all injection points
+* `BEANS`
+** `Collection<BeanInfo>` containing all beans
+* `REMOVED_BEANS`
+** `Collection<BeanInfo>` containing all the removed beans; see <<remove_unused_beans>> for more information
+* `OBSERVERS`
+** `Collection<ObserverInfo>` containing all observers
+* `SCOPES`
+** `Collection<ScopeInfo>` containing all scopes, including custom ones
+* `QUALIFIERS`
+** `Map<DotName, ClassInfo>` containing all qualifiers
+* `INTERCEPTOR_BINDINGS`
+** `Map<DotName, ClassInfo>` containing all interceptor bindings
+* `STEREOTYPES`
+** `Map<DotName, ClassInfo>` containing all stereotypes
```

Comment:
Something I found during the CDI Lite work: this is actually incorrect (and not caught by javac :-) ), the `Map` stored in the context is in fact `Map<DotName, StereotypeInfo>`. Which actually makes more sense.

Reply:
Ok. This part was CTRL+C and CTRL+V. At least we can fix it right away ;-)